src/commands/createStaticWebApp/ApiLocationStep.ts

```diff
@@ -4,19 +4,27 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { AzureWizardPromptStep } from "vscode-azureextensionui";
-import { apiSubpathSetting, defaultApiName } from "../../constants";
+import { apiSubpathSetting, defaultApiName, skipForNowQuickPickItem } from "../../constants";
 import { ext } from "../../extensionVariables";
+import { getGitTreeQuickPicks } from "../../utils/gitHubUtils";
 import { localize } from "../../utils/localize";
+import { nonNullProp } from "../../utils/nonNull";
 import { getWorkspaceSetting } from "../../utils/settingsUtils";
 import { addLocationTelemetry } from "./addLocationTelemetry";
 import { IStaticWebAppWizardContext } from "./IStaticWebAppWizardContext";
 
 export class ApiLocationStep extends AzureWizardPromptStep<IStaticWebAppWizardContext> {
     public async prompt(wizardContext: IStaticWebAppWizardContext): Promise<void> {
-        wizardContext.apiLocation = (await ext.ui.showInputBox({
-            value: getWorkspaceSetting(apiSubpathSetting, wizardContext.fsPath) || defaultApiName,
-            prompt: localize('apiLocation', "Enter the location of your Azure Functions code (Leave blank for no Azure Functions code)"),
-        })).trim();
+        const placeHolder: string = localize('apiLocation', "Enter the location of your Azure Functions code");
+        const input: string = (await ext.ui.showQuickPick(
+            await getGitTreeQuickPicks(
+                nonNullProp(wizardContext, 'gitTreeData'),
+                getWorkspaceSetting(apiSubpathSetting, wizardContext.fsPath),
+                true),
+            { placeHolder, suppressPersistence: true })).label.trim();
```

Comment:
You should always use `data` instead of `label` for storing the actual value. Too many bugs are caused by a display value (label) being slightly different than the actual value (data). That would mean you don't have to check for `skipForNowQuickPickItem.label` below since you can just make `data` whatever it needs to be in that case